### PR TITLE
train_adversarial: Remove custom plotting

### PR DESCRIPTION
**Pull request by shwang**
I decided to remove custom plotting because:
(1) Now that we plot disc loss and other things in Tensorboard, the custom plotter is almost entirely redundant.
(2) Also there seems to be a bug in master (maybe related to our complicated reward/observation normalization copying scheme) that is under-reporting episode rewards in our custom plots. EG: TensorBoard shows Cartpole reaching 500, but our custom plots have reward less than 200.

Also moves the training loop to AdversarialTrainer.train() since we only need a simple callback after this simplification.

### CODE REVIEW

**Comment by shwang** [comment]
Possible hazard: These custom plots helped us figure out that our Stable Baselines plots were faulty, letting them go, especially before we get them in a working state that corroborates the stable baselines plots can be sort of testy.

However, we can tell from the form of the Tensorboard plots that the early-end-of-episode bug from before is fixed. For example, the 500 reward of the gray Cartpole run would be impossible under the previous regime because the batch size doesn't accommodate such a long episode:

image

**Comment by AdamGleave** [comment]
> Possible hazard: These custom plots helped us figure out that our Stable Baselines plots were faulty, letting them go, especially before we get them in a working state that corroborates the stable baselines plots can be sort of testy.

I would like to understand why they disagree. No need to fix the bug if it's hard, but not knowing what's wrong is concerning.

File: src/imitation/algorithms/adversarial.py [hunk 1]
```diff
@@ -225,6 +226,26 @@ def train_disc_step(self, *,
         logger.logkv(k, v)
       logger.dumpkvs()
 
+  def eval_disc_loss(self, **kwargs) -> float:
+    """Evaluates the discriminator loss.
+
+    The generator rollout parameters of the form "gen_*" are optional,
```
Comment (AdamGleave) [inline src/imitation/algorithms/adversarial.py]: there's only one keyword argument beginning with `gen_*` documented below? confused

File: src/imitation/algorithms/adversarial.py [hunk 2]
```diff
@@ -243,35 +264,41 @@ def train_gen(self, total_timesteps: Optional[int] = None, callback=None):
 
     with logger.accumulate_means("gen"):
       self.gen_policy.set_env(self.venv_train_norm_buffering)
-      # TODO(adam): learn was not intended to be called for each training batch
```
Comment (AdamGleave) [inline src/imitation/algorithms/adversarial.py]: if we're removing this warning comment, feel like we should also point to the fixed branch of stable baselines in dependency (or just wait until it gets merged to master)

File: src/imitation/algorithms/adversarial.py [hunk 3]
```diff
@@ -243,35 +264,41 @@ def train_gen(self, total_timesteps: Optional[int] = None, callback=None):
...
-    Args:
-      gen_samples (Optional[rollout.Transitions]): Same as in `train_disc_step`.
-      expert_samples (Optional[rollout.Transitions]): Same as in
-        `train_disc_step`.
-
-    Returns:
-      The total cross-entropy error in the discriminator's classification.
+  def train(self,
+            total_timesteps: int,
+            callback: Optional[Callable[[bool], None]] = None,
```
Comment (AdamGleave) [inline src/imitation/algorithms/adversarial.py]: why is this `bool`? should it not be `int`?

File: src/imitation/algorithms/adversarial.py [hunk 4]
```diff
@@ -243,35 +264,41 @@ def train_gen(self, total_timesteps: Optional[int] = None, callback=None):
...
-    is generated on the fly.
-
-    Args:
-      gen_samples (Optional[rollout.Transitions]): Same as in `train_disc_step`.
-      expert_samples (Optional[rollout.Transitions]): Same as in
-        `train_disc_step`.
-
-    Returns:
-      The total cross-entropy error in the discriminator's classification.
+  def train(self,
```
Comment (AdamGleave) [inline src/imitation/algorithms/adversarial.py]: :+1: nice we have more of the training logic in this class now

**Comment by shwang** [comment]
> No need to fix the bug if it's hard, but not knowing what's wrong is concerning.

That's fair. I'll take another look into this...

File: src/imitation/algorithms/adversarial.py (hunk 2 above)
Comment (shwang) [reply src/imitation/algorithms/adversarial.py]: I was just thinking about making setup.py point to the Runner patch (had an error earlier this week where I installed SB without my patch).

File: src/imitation/algorithms/adversarial.py (hunk 1 above)
Comment (shwang) [reply src/imitation/algorithms/adversarial.py]: Hmm I thought I had deleted this part of the documentation a while ago. Maybe I made a bad merge sometime back.

File: src/imitation/algorithms/adversarial.py (hunk 2 above)
Comment (shwang) [reply src/imitation/algorithms/adversarial.py]: Looks like it's fine for us to go ahead and merge the Runner patch after your review+approval. Let's merge this PR after that change.

**Comment by shwang** [comment]
Actually, plots are fine!

I did find a custom plotting bug involving stale observation normalization though. Didn't seem to make a big difference on reporting cartpole before and after fixing it.

image

image

**Comment by AdamGleave** [comment]
> Actually, plots are fine!
> 
> I did find a custom plotting bug involving stale observation normalization though. Didn't seem to make a big difference on reporting cartpole before and after fixing it.
> 
> image
> 
> image

I assume the custom plots are for the green-blue and dark-orange seeds in the TB plot? If so agree looks like they match closely (might want to reduce smoothing in TB to allow more direct comparison).

**Comment by AdamGleave** [comment]
One of the `codecov` uploads failed on CI, I'm rerunning that job, think checks should be green after that.

**Comment by shwang** [comment]
> I assume the custom plots are for the green-blue and dark-orange seeds in the TB plot?

Yes, IIRC it's green-blue and dark-orange.
